agents/World.java

```diff
@@ -119,11 +131,12 @@ public Continent getContinent() {
         }
     }
 
-    public class City {
+    public class City extends Loggable {
```

Comment:
Same here. Let's not use inheritance if it doesn't reflect the domain problem

Make `Loggable` a `Logger` class and put it under `//common`. Then instantiate it within `City` like this:

```
class City {
  private Logger logger = new Logger();
  
}
```